### DISCUSSION TITLE
Luba with game controller

### DISCUSSION THREAD

**Discussion by shagadeli** [mikey0000/PyMammotion#9] [Q&A] [answered]
Reached this when looking for an integration of Luba to home assistant! 
Unfortunately I don't really know how to help, but read about the game controller capability and got interested!
Would you be able to give some hints about whats needed to accomplish that?

Thanks

**Comment by mikey0000** [comment]
Well it already works, what it needs is some tidying up, maybe a GUI and some tweaks. Heck you could write a desktop app. I am trying to find more time to work on the WiFi side of things for a home assistant integration so I'm not working directly on the controller stuff, but I mean I've driven Luba around etc with a controller so no issues there. Including mowing and so on over Bluetooth. Technically speaking you can send commands over WiFi but that would be very laggy if it even worked at all and also untested right now.

Couple things here. 1. Is making the api interface easier to deal with so any integration whether it be joystick or HA can use it.

2. Is what is your goal exactly? Desktop app? just running a command line function?

23/02/2024 1:53:04 pm shagadeli ***@***.***>:

**Reply by shagadeli** [reply]
What I'm actually looking forward to is to use it as a home assistant integration to control the mower pretty much like a robot vaccum cleaner (just like I do with the worx landroid)
This just seem to be quite far from reality so I got excited with the idea of manually control it with a gamepad. Right now the app is just too buggy to manually mow as its dropping the bluetooth every 10-30s, hopping for an update from mammotion soon.

I also hope they release a MQTT support from them so it would be easier to get status update.

What hardware is needed to use the game controller?

**Comment by mikey0000** [comment]
PlayStation or Xbox controller/any controller really

Im working on the home assistant integration, can actually do MQTT now but getting the keys requires a rooted android device. I'm trying to work out what the secret is for the initial two requests signature. I have a plan though and am getting closer.

29/02/2024 9:13:46 am shagadeli ***@***.***>:

**Reply by shagadeli** [reply]
Thats such great news! 

Regarding the game controller, besides the controller itself whats needed? How are the commands transmitted to the mower? Where is the code uploaded?

**Reply by mikey0000** [reply]
See the code under tests, should be one for controllers

**Comment by mikey0000** [comment, marked as answer]
This repo, look at the test folder there should be sample code in there, can run it from the command line and it will look for a Bluetooth connection to a Luba and attempt to connect.

Ive been refactoring code so some things might be referenced wrong but otherwise good, poetry. Is the tool to use

29/02/2024 10:57:52 am shagadeli ***@***.***>:
